cpp/src/gandiva/function_registry_string.cc

```diff
@@ -100,6 +100,10 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
                      kResultNullIfNull, "gdv_fn_like_utf8_utf8",
                      NativeFunction::kNeedsFunctionHolder),
 
+      NativeFunction("like", {}, DataTypeVector{utf8(), utf8(), int8()}, boolean(),
```

Comment:
why not provide utf8 instead of int8? or it won't work with multibyte utf8 character.

Reply:
Int8 is used here because function [SqlLikePatternToPcre](https://github.com/apache/arrow/blob/master/cpp/src/gandiva/regex_util.h#L33) only supports a single-byte ASCII char as the escape character. 
The change of multibyte escape character may not trivial. Maybe we can create a new issue and another PR to implement it?

Reply:
Sure creating a different ticket to support escaping multibyte char seems fine. I still think you should provide utf8 in the signature instead of int8 and add a check that it is of length 1, since the argument is string type. Or else it won't work when you call the function from java

Reply:
The code is updated and the int8 type is replaced with utf8 type.